numpy/core/tests/test_simd.py

```diff
@@ -632,6 +699,24 @@ def test_arithmetic_reduce_sum(self):
         vsum = self.sum(vdata)
         assert vsum == data_sum
 
+    def test_mask_conditional(self):
+        vdata_a = self.load(self._data())
+        vdata_b = self.load(self._data(reverse=True))
+        true_mask  = self.cmpeq(self.zero(), self.zero())
+        false_mask = self.cmpneq(self.zero(), self.zero())
+
+        data_sub = self.sub(vdata_b, vdata_a)
+        ifsub = self.ifsub(true_mask, vdata_b, vdata_a, vdata_b)
+        assert ifsub == data_sub
+        ifsub = self.ifsub(false_mask, vdata_a, vdata_b, vdata_b)
+        assert ifsub == vdata_b
+
+        data_add = self.add(vdata_b, vdata_a)
+        ifadd = self.ifadd(true_mask, vdata_b, vdata_a, vdata_b)
+        assert ifadd == data_add
+        ifadd = self.ifadd(false_mask, vdata_a, vdata_b, vdata_b)
+        assert ifadd == vdata_b
+
```

Comment:
@mattip,

> I meant adding some kind of comment or other notation in the test file itself to make it clear what intrinsics each test is checking.

please take a look here, we are here testing mask conditional operations on several SIMD extensions with different data types. in the same time, we count on another NPYV intrinsics for generating the testing data and args in order to shrink the amount of code.

in other words, the code here equivalent to the following code except it testing all vector data types on all supported SIMD targets. sweet isn't it?:

<details>

<summary>following code</summary>

```Python

from numpy.core._simd import targets
npyv = targets["baseline"]
vdata_a = npyv.load_u8(range(npyv.nlanes_u8))
vdata_b = npyv.load_u8(reversed(range(npyv.nlanes_u8)))

true_mask  = npyv.cmpeq_u8(npyv.zero_u8(), npyv.zero_u8())
false_mask = npyv.cmpneq_u8(npyv.zero_u8(), npyv.zero_u8())

data_sub = npyv.sub_u8(vdata_b, vdata_a)
ifsub = npyv.ifsub_u8(true_mask, vdata_b, vdata_a, vdata_b)
assert ifsub == data_sub

ifsub = npyv.ifsub_u8(false_mask, vdata_a, vdata_b, vdata_b)
assert ifsub == vdata_b

data_add = npyv.add_u8(vdata_b, vdata_a)
ifadd = npyv.ifadd_u8(true_mask, vdata_b, vdata_a, vdata_b)
assert ifadd == data_add

ifadd = npyv.ifadd_u8(false_mask, vdata_a, vdata_b, vdata_b)
assert ifadd == vdata_b

```

</details>


So I wonder, How can we add notations or special marks? we can replace parm name `self` to `npyv`
but would it help? I guess not. The only solution in my head right now is adding a docstring for each testing function clarifying the following:

* the place where these intrinsics are defined in the source
* the signatures of each one of them

Reply:
Let's start with a hand written docstring that just mentions "testing `cmpeq_u8`, `cmpneq_u8`, ..." for every new test in this PR. No need to do more than that now. Then the reviewers can see which intrinsics have been added and that there are tests that cover them.